src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,161 @@
+package core.basesyntax.service;
+
+import static core.basesyntax.service.RegistrationServiceImpl.MAX_AGE;
+import static core.basesyntax.service.RegistrationServiceImpl.MIN_AGE;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationService registrationService;
+    private static StorageDao storageDao;
+    private static User testUser;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+        testUser = new User();
+    }
+
+    @BeforeEach
+    void setUp() {
+        testUser.setLogin("Elvis");
+        testUser.setAge(MIN_AGE);
+        testUser.setPassword("_elviS");
+    }
+
+    @Test
+    void register_nullUser_notOk() {
+        User nullUser = null;
+        assertThrows(RuntimeException.class, () -> registrationService.register(nullUser));
+    }
+
+    @Test
+    void register_nullLogin_notOk() {
+        testUser.setLogin(null);
+        assertThrows(RuntimeException.class, () -> registrationService.register(testUser));
+    }
+
+    @Test
+    void register_emptyLogin_notOk() {
+        testUser.setLogin("");
+        assertThrows(RuntimeException.class, () -> registrationService.register(testUser));
+    }
+
+    @Test
+    void register_alreadyExistsUser_NotOK() {
+        storageDao.add(testUser);
+        testUser.setLogin("Elvis");
+        testUser.setAge(MAX_AGE);
+        testUser.setPassword("123456");
+        assertThrows(RuntimeException.class, () -> registrationService.register(testUser));
+    }
+
+    @Test
+    void register_validMaxAgeUser_Ok() {
+        testUser.setLogin("ElvisMaxAge");
+        testUser.setAge(MAX_AGE);
+        testUser.setPassword("123456");
+        registrationService.register(testUser);
+        assertTrue(MAX_AGE > MIN_AGE);
+        User findUser = storageDao.get(testUser.getLogin());
+        assertEquals(testUser.getLogin(), findUser.getLogin());
+        assertEquals(testUser.getAge(), findUser.getAge());
+        assertEquals(testUser.getPassword(), findUser.getPassword());
```

Comment:
```suggestion
        User actual = registrationService.register(testUser);
        assertTrue(MAX_AGE > MIN_AGE);
        assertEquals(testUser, actual);
```

we want to test work of method `register`, not `storageDao.get()`. Same in other tests

Reply:
Thank you. Yes, I agree with you. I just thought that if the user is successfully registered, then he will be in the storage:)